examples/tools/autotools/autotoolstoolchain/string_formatter/conanfile.py

```diff
@@ -0,0 +1,29 @@
+from conan import ConanFile
+from conan.tools.gnu import AutotoolsToolchain, PkgConfigDeps, Autotools
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.microsoft import unix_path
+import os
+
+
+class StringFormatterConanFile(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
+
+    def requirements(self):
+        self.requires("fmt/9.1.0")
+
+    def build_requirements(self):
+        if self.settings.os == "Windows":
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
```

Comment:
I think we want to start deprecating this idiom, I know it is popular in ConanCenter, but this is for 2.0 and Conan 2.0 has better tools like ``[system_tools]``.
I think for this particular example:
- I would avoid build-requires in the recipe, as it should work quite out of the box in Linux and Mac systems, without build_requires.
- If necessary, I would put some build-requires in the Windows ``profile``. Maybe this is possible?